### ISSUE TITLE
Issue with xbox controller Mac M1

### ISSUE THREAD

**Issue by RudyProme** [unknownskl/xbox-xcloud-client#469]
hello, my Xbox controller is not working on the app with Mac M1. My Xbox controller is working in Steam big picture, also in the link https://gamepad-tester.com/ from Chrome or Safari. But it doesn't recognize on the App (beta 2.0 and 1.2)

**Comment by MisterAnderson91** [comment]
Which xbox controller are you using, xbox 360, xbox one original (wireless receiver or usb), xbox one bluetooth, or xbox series?

Make sure Steam isn't running in case it is capturing the inputs.

I have personally used an xbox series controller over bluetooth on an M1 Mac with no issues.

**Comment by RudyProme** [comment]
Hi, I confirm Steam is not running. I tried with Dualsense and Xbox Series Elite 2 controllers, both over Bluetooth

**Comment by unknownskl** [comment]
Can you check in the system settings -> security & privacy -> privacy tab -> input monitoring and check if the app is added there? If not, try to add it and restart the application and check if it works then.

**Comment by RudyProme** [comment]
Yes the app is already added there as it asked when you first start the app

**Comment by Reaster0** [comment]
hi, mac m1 user here, i don't have any issue with it, maybe try to connect the controller after the start of the stream?

**Comment by kaputt2k** [comment]
running into the same issue here.
M2 Macbook air, xbox wireless controller via bluetooth. 

It worked like acharm for a month, but suddenly stopped working. The Gamepad still works is i use xcloud via any other browser, but is not detected in greenlight.  no clue how to fix this. 
<img width="643" alt="image" src="https://user-images.githubusercontent.com/45491914/197417970-029c558d-d2ef-47f6-8938-c32f19206501.png">
<img width="305" alt="image" src="https://user-images.githubusercontent.com/45491914/197418024-cb7192d3-f145-4196-a5b0-32fcd5b525fb.png">

**Comment by RudyProme** [comment]
Still with the issue. Same thing as kaputt2k

**Comment by markkrueg** [comment]
I have the same problem. I'm using an Xbox Elite Wireless Controller Series 2 on macOS Ventura 13.0. MacBook Pro M1 Max. I have tried connecting with both bluetooth and USB.

**Comment by unknownskl** [comment]
The new v0.2.0-beta2 has some fixes which didnt display the controllers properly on the settings tab. Can you check if the problem still persists with the new version?

**Comment by RudyProme** [comment]
Hi, I just tested it again with the new beta version and I still got the problem. I tried with Dualsense controller (bluetooth and USB) and with Xbox Series Elite 2 controller.

**Comment by unknownskl** [comment]
Did you try to input any buttons? Due to how electron works under the hood, the gamepad will only be shown when it received at least 1 button press from the gamepad. My Xbox Series Elite 2 works fine over here, including the original Xbox One controllers.

**Comment by unknownskl** [comment]
Gotcha. Is that on 2.0.0-beta1 or 2.0.0-beta2?

**Comment by RudyProme** [comment]
this is on the beta2

**Comment by bradarl** [comment]
I don't issues with my Xbox Wireless Controller (standard, series X|S) but I do have issues with my Elite Series 2 (it doesn't work) 

I can, however, use the Elite Series 2 controller with the Xbox app on a Windows 11 VM (Parallels) when it is connected via USB (not Bluetooth)

Machine: Macbook Pro M1
OS: Ventura 16.0.1

**Comment by quidnu** [comment]
Same problem with 8bitdo Ultimate bluetooth controller
Input monitoring enabled for Greenlight
Controller is not listed in detected controllers in Greenlight.
Works fine in "Game Controller Tester" in the App Store.

Greenlight beta2
MacBook Air M1
OS: Ventura 13.0 (22A380)

**Comment by quidnu** [comment]
I did some further testing and found that the controller works on https://gamepad-tester.com in Safari but not Chrome, Firefox nor Edge so I suspect the issue is there.

**Comment by bradarl** [comment]
The controller works for me in Firefox but not the beta app.

**Comment by quidnu** [comment]
> The controller works for me in Firefox but not the beta app.
> […](#)

Electron uses Chrome, does it work in Chrome?

**Comment by bradarl** [comment]
Confirmed, Xbox Elite 2 controller works wirelessly with Google Chrome.

OS: MacOS 13.0.1
Chrome: 108.0.5359.94

**Comment by MisterAnderson91** [comment]
> I did some further testing and found that the controller works on https://gamepad-tester.com in Safari but not Chrome, Firefox or Edge so I suspect the issue is there.

I have found that joy-cons don't get detected in any browser but Safari, but still work fine in Greenlight for whatever reason, so it may not mean anything.

**Comment by Julepo1** [comment]
Same problem here. 

Xbox elite V2 works properly in chrome and safari
Input monitoring enabled for Greenlight
Controller is not listed in detected controllers in Greenlight Beta 2, and doesn't work when trying to stream
Mac book Air M2
MAC OS 13.0.1

**Comment by ghost** [comment]
Issue still remains for me in Beta 12. Gamepad not detected. 
MacBook Pro M1 Pro 
Mac OS 14

**Comment by slapaglia** [comment]
Same issue Mac
Studio M2
Ventura

**Comment by apirkle** [comment]
Same issue on a MBP M3 / Sonoma 14.2

Input Monitoring is enabled for Greenlight in System Preferences.
Web test on gamepad-tester.com works.
No response in Greenlight app.

Followed suggestion in #1002 to launch Greenlight from Terminal. Ran as normal user (not sudo) via /Applications/Greenlight.app/Contents/MacOS/Greenlight

This resulted in a prompt to enable Input Monitoring privs for Terminal. On restart of Terminal + relaunch of Greenlight from Terminal, the controller was detected and started working normally.
